### DISCUSSION TITLE
how to import installed packages into a CMake project

### DISCUSSION THREAD

**Discussion by ac169** [microsoft/vcpkg#42670] [Q&A] [answered]
```
Total install time: 549 us
The package fmt provides CMake targets:

    find_package(fmt CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fmt::fmt)

    # Or use the header-only version
    find_package(fmt CONFIG REQUIRED)
    target_link_libraries(main PRIVATE fmt::fmt-header-only)
```

most packages have the above prompt to import the CMakeLists.txt file after installation, while a small number of packages do not have this prompt after installation. So how to import such packages, for example: **vcpkg install wepoll:x64-windows**

```
Computing installation plan...
The following packages are already installed:
    wepoll:x64-windows@1.5.8#3
wepoll:x64-windows is already installed
```

**find_package(wepoll REQUIRED)**

```
By not providing "Findwepoll.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "wepoll", but
  CMake did not find one.

  Could not find a package configuration file provided by "wepoll" with any
  of the following names:

    wepollConfig.cmake
    wepoll-config.cmake
```

**Comment by LiAuTraver** [comment]
not sure what you mean -- did you mean use the port in `CMakeLists`? 
```powershell
 ❯ vcpkg integrate install
Applied user-wide integration for this vcpkg root.
CMake projects should use: "-DCMAKE_TOOLCHAIN_FILE={your vcpkg's root path}/scripts/buildsystems/vcpkg.cmake"

All MSBuild C++ projects can now #include any installed libraries. Linking will be handled automatically. Installing new libraries will make them instantly available.
```
more info see [documentation](https://learn.microsoft.com/en-us/vcpkg/get_started/get-started?pivots=shell-powershell#4---build-and-run-the-project).

**Reply by ac169** [reply]
How to introduce the vcpkg environment in CMakeLists is no problem. My problem is that there is no prompt after installation to correctly configure the package name parameters in find_package or target_link_libraries.

**Reply by aureliencc** [reply, marked as answer]
I went into the same situation with libmupdf.

You can try to include the library with : 

```
find_library(WEPOLL_LIBRARY wepoll)
target_link_libraries(${YOUR_PROJECT_NAME} PRIVATE ${WEPOLL_LIBRARY})
```

Beware that it does not handle dependencies.

**Reply by ac169** [reply]
> I went into the same situation with libmupdf.
> 
> You can try to include the library with :
> 
> ```
> find_library(WEPOLL_LIBRARY wepoll)
> target_link_libraries(${YOUR_PROJECT_NAME} PRIVATE ${WEPOLL_LIBRARY})
> ```
> 
> Beware that it does not handle dependencies.

thank you!

**Comment by dg0yt** [comment]
CMake (config) packages should be provided by upstream maintainers, not by ports. For some packages, there are also Find modules provided by CMake.

For some packages, vcpkg contributors add unofficial CMake config. This is useful when features and dependencies are complex.  (libmupdf would be a good candidate.)

For everything else, there is plain `find_path` and `find_library`. Which is also what Find modules do.

**Reply by ac169** [reply]
> CMake (config) packages should be provided by upstream maintainers, not by ports. For some packages, there are also Find modules provided by CMake.
> 
> For some packages, vcpkg contributors add unofficial CMake config. This is useful when features and dependencies are complex. (libmupdf would be a good candidate.)
> 
> For everything else, there is plain and . Which is also what Find modules do.`find_path``find_library`

There are many ways to use the third package, such as importing the source code directly into your own project. There is more confusion about the friendliness of new users and the inconsistency of introducing standards. What is the point of using vcpkg?
